src/core/reactor.cc

```diff
@@ -2218,6 +2219,45 @@ void reactor::kill(pid_t pid, int sig) {
     ret.throw_if_error();
 }
 
+future<group_details> reactor::grp_detail(std::string_view groupname) noexcept {
+    return futurize_invoke([groupname, this] {
+        return _thread_pool->submit<syscall_result_extra<struct group>>([groupname = sstring(groupname)] {
+            struct group *gg = ::getgrnam(groupname.c_str());
```

Comment:
please set `errno` to zero before calling this API. as we are going to check `errno` in `wrap_syscall()` if the retval is -1, see https://man7.org/linux/man-pages/man3/getgrnam.3.html#RETURN_VALUE

Reply:
Please use the thread-safe getgrnam_r() instead of getgrnam().

Reply:
I looked into using `getgrnam_r()` instead of `getgrname` , the docs for `getgrnam_r` says
```
On success, getgrnam_r() and getgrgid_r() return zero, and set *result to grp. If no matching group record was found, these functions return 0 and store NULL in *result. In case of error, an error number is returned, and NULL is stored in *result.
```
I was going to take the following approach here:

```
future<std::optional<struct group>> reactor::getgrnam(std::string_view name, char *buf, size_t buflen) noexcept {
    return futurize_invoke([name, buf, buflen, this] {
        return _thread_pool->submit<syscall_result_extra<std::optional<struct group>>>([name = sstring(name), buf, buflen] {
            struct group grp;
            struct group *result;
            int ret = ::getgrnam_r(name.c_str(), &grp, buf, buflen, &result);
            if (!result) {
                return wrap_syscall(ret, std::optional<struct group>(std::nullopt));
            }
            return wrap_syscall(ret, std::optional<struct group>(*result));
        }).then([name = sstring(name)] (syscall_result_extra<std::optional<struct group>> gr) {
            if (gr.result != 0) {
                throw std::system_error(gr.result, std::system_category());
            }
            return make_ready_future<std::optional<struct group>>(std::optional<struct group>(std::move(gr.extra)));
        });
    });
}
```

since I am returning an optional\<group\> to the user, to communicate the system error number to the caller, `throw std::system_error(gr.result, std::system_category())` that way the user can handle these errors on their end
```
ENOENT or ESRCH or EBADF or EPERM ...
```

Thoughts..?

Reply:
couple things:

1. you missed https://github.com/scylladb/seastar/pull/2314#discussion_r1660393108
2. please check `ret` first
3. might want to s/gr/sr/ or just s/gr/ret/ as `group` is implied here
4. please throw `std::error_code(gr.error, std::system_category())`

Reply:
and, please just send the updated pull request instead of asking for confirmation from reviewers.